app/src/app/townhall/page.tsx

```diff
@@ -440,6 +448,15 @@ const KageChallenge: React.FC<{
   );
   const isAtWar = activeVillageWars && activeVillageWars.length > 0;
 
+  // Calculate daily locked time information
+  const dailyLockedTimeSeconds = dailyLockedTimeData?.dailyLockedTimeSeconds ?? 0;
+  const maxDailySeconds = KAGE_CHALLENGE_MAX_DAILY_LOCKED_HOURS * 60 * 60;
+  const remainingSeconds = Math.max(0, maxDailySeconds - dailyLockedTimeSeconds);
+  const remainingHours = Math.floor(remainingSeconds / 3600);
+  const remainingMinutes = Math.floor((remainingSeconds % 3600) / 60);
```

Comment:
We have a convenience function for this `getDaysHoursMinutesSeconds` in `app/src/utils/time.ts`